src/View/Widget/DateTimeWidget.php

```diff
@@ -491,36 +531,63 @@ protected function _secondSelect($options, $context)
      */
     protected function _meridianSelect($options, $context)
     {
+        $date = new Time('09:00:00', $options['localization']['timezone']);
         $options += [
             'name' => '',
             'val' => null,
-            'options' => ['am' => 'am', 'pm' => 'pm']
+            'options' => [
+                'am' => $date->i18nFormat('a'),
+                'pm' => $date->addHours(12)->i18nFormat('a')
+            ]
         ];
+
+        unset($options['localization']);
         return $this->_select->render($options, $context);
     }
 
     /**
+     * Generates a localization hidden input
+     *
+     * @param array $options The options to generate a hidden input
+     * @param \Cake\View\Form\ContextInterface $context The current form context.
+     * @return string
+     */
+    protected function _localizationSelect($options, $context)
+    {
+        $out = '';
+        $template = '<input name="%s[%s]" value="%s" type="hidden">';
```

Comment:
All the templates are available in the StringTemplate object (`$this->_templates`)